### PR TITLE
Import mount points from plain encryptions

### PR DESCRIPTION
**Pull request by joseivanlopez**
## TODO

* Merge this branch into https://github.com/yast/yast-storage-ng/pull/960.
* Adapt to the library API for creating plain encryption (see comments in this PR).
* Add unit tests.
* Manual testing.

---

## Problem

When using the "Import Mount Points" option in the Expert Partitioner, mount points from plain encryption devices are not imported (typically for encrypted swap with random password).

* Part of PBI: https://trello.com/c/z9KlLE1m/1243-5-sles15-p3-1088641-partitioner-encrypted-swap-with-random-password

## Solution

Import mount points even when the plain encryption device was not probed.

## Testing

- *Added a new unit test*
- *Tested manually*

## Screenshots

*If the fix affects the UI attach some screenshots here.*

### CODE REVIEW

File: src/lib/y2partitioner/actions/controllers/fstabs.rb [hunk 1]
```diff
@@ -441,11 +449,57 @@ def devicegraph_with_crypttab_names(devicegraph)
...
+        #
+        # @param device [Y2Storage::Encryption]
+        # @param devicegraph [Y2Storage::Devicegraph]
+        #
+        # @return [Y2Storage::Encryption]
+        def create_plain_encryption(device, devicegraph)
+          blk_device = devicegraph.find_device(device.blk_device.sid)
+          blk_device.remove_descendants
+
+          blk_device.encrypt(device.crypttab_name, device.password, Y2Storage::EncryptionType::PLAIN)
```
Comment (joseivanlopez) [inline src/lib/y2partitioner/actions/controllers/fstabs.rb]: Replace `::PLAIN` with the new encryption type provided by the library.

File: src/lib/y2storage/blk_device.rb [hunk 2]
```diff
@@ -293,9 +294,11 @@ def encrypted?
     #
     # @param dm_name [String, nil] DeviceMapper table name of the new device
     # @param password [String, nil] password of the new device
+    # @param type [EncryptionType]
+    #
     # @return [Encryption]
-    def encrypt(dm_name: nil, password: nil)
-      enc = create_encryption(dm_name || "")
+    def encrypt(dm_name: nil, password: nil, type: EncryptionType::LUKS1)
```
Comment (joseivanlopez) [inline src/lib/y2storage/blk_device.rb]: Adapt this method to the new library API for creating plain encryption.

File: src/lib/y2storage/encryption.rb [hunk 3]
```diff
@@ -232,13 +232,28 @@ def save_crypttab_names(devicegraph, crypttab)
...
 
-        device.encryption.crypttab_name = entry.name
+        return unless device
+
+        if device.encrypted?
+          device.encryption.crypttab_name = entry.name
+        elsif entry.random_password?
+          # Considering plain encryption when using random password
+          device.remove_descendants
+          device.encrypt(entry.name, entry.password, EncryptionType::PLAIN)
```
Comment (joseivanlopez) [inline src/lib/y2storage/encryption.rb]: Replace `::PLAIN` with the new encryption type provided by the library.

**Comment by dgdavid** [comment]
Just for the record, to avoid those merge commits in #960 I "undid" them and 


* Downloaded the format patch of the commit      
   > https://github.com/yast/yast-storage-ng/commit/5d70c752692371597cc0a76a3c3c4041972dbf39.patch

* Edited it removing the conflicts

* Applied it via `git am`

  >  git am -i --reject 5d70c75.patch

* Forced-pushed the changes

So, merge commits are no longer there.

**Pull request merged**